src/profile/external-identities/FrameworkServiceLink.tsx

```diff
@@ -0,0 +1,34 @@
+import { ClickableProps } from '@terra-ui-packages/components';
+import React, { ReactNode } from 'react';
+import { ButtonPrimary, Link } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { Ajax } from 'src/libs/ajax';
+import { withErrorReporting } from 'src/libs/error';
+import * as Utils from 'src/libs/utils';
+
+interface FrameworkServiceLinkProps extends ClickableProps {
+  linkText: string;
+  providerKey: string;
+  redirectUrl: string;
+  button?: boolean;
+}
+
+export const FrameworkServiceLink = (props: FrameworkServiceLinkProps): ReactNode => {
+  const { linkText, providerKey, redirectUrl, button = false, ...clickableProps } = props;
+  const Component = button ? ButtonPrimary : Link;
+  const style = button ? {} : { display: 'inline-flex', alignItems: 'center' };
+  return (
+    <Component
+      onClick={withErrorReporting('Error getting Fence Link', async () => {
+        const result = await Ajax().User.getFenceAuthUrl(providerKey, redirectUrl);
+        window.open(result.url, Utils.newTabLinkProps.target, 'noopener,noreferrer');
+      })}
+      style={style}
+      /* eslint-disable-next-line react/jsx-props-no-spreading */
+      {...clickableProps}
```

Comment:
This is a nit, and practically probably doesn't matter, but a consumer _could_ pass an `onClick` handler. As is, it would override the desired `onClick` handler that opens the auth URL.

You could avoid this by placing `onClick` after `...clickableProps`: 
```suggestion
      style={style}
      /* eslint-disable-next-line react/jsx-props-no-spreading */
      {...clickableProps}
      onClick={withErrorReporting('Error getting Fence Link', async () => {
        const result = await Ajax().User.getFenceAuthUrl(providerKey, redirectUrl);
        window.open(result.url, Utils.newTabLinkProps.target, 'noopener,noreferrer');
      })}
```

And/or, you could remove `onClick` from the props type.
```ts
interface FrameworkServiceLinkProps extends Omit<ClickableProps, 'onClick'> {
```

Reply:
I like removing `onClick` from the prop types. That way, it'll be a louder error if a consumer tries to pass it in, instead of it just silently ignoring the consumer-provided handler.